src/main/java/org/folio/inventory/services/ItemService.java

```diff
@@ -0,0 +1,88 @@
+package org.folio.inventory.services;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.inventory.storage.external.CqlQuery.exactMatch;
+import static org.folio.inventory.storage.external.CqlQuery.match;
+import static org.folio.inventory.storage.external.CqlQuery.notEqual;
+import static org.folio.inventory.storage.external.Limit.one;
+import static org.folio.inventory.storage.external.Offset.noOffset;
+import static org.folio.inventory.support.JsonArrayHelper.toList;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.inventory.common.WebContext;
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemCollection;
+import org.folio.inventory.domain.items.ItemStatusName;
+import org.folio.inventory.domain.view.request.RequestStatus;
+import org.folio.inventory.domain.view.request.StoredRequestView;
+import org.folio.inventory.storage.external.Clients;
+import org.folio.inventory.storage.external.CollectionResourceRepository;
+import org.folio.inventory.storage.external.CqlQuery;
+import org.folio.inventory.validation.ItemMarkAsWithdrawnValidators;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ItemService {
```

Comment:
Could this service be specific only to withdrawing an item? What makes it more general than that?

Reply:
I can rename it to `MoveItemToStatusService`, in this case it can be reused for similar updates (for example for missing items).

What do you think?

Reply:
Are we intending to try to reuse this for other actions?

Personally, I prefer to defer trying to generalise something (especially to reduce duplication) till the system has multiple examples to guide the generalisation.

Reply:
I'm fine to make it specific for withdraw case. Will rename it to `WithdrawItemService`.